src/pb/ipns.proto.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const protons = require('protons')
+
+/* eslint-disable no-tabs */
+const message = `
+message IpnsEntry {
+  enum ValidityType {
+		EOL = 0; // setting an EOL says "this record is valid until..."
+	}
+
+  required bytes value = 1;
+	required bytes signature = 2;
+
+	optional ValidityType validityType = 3;
+	optional bytes validity = 4;
+
+	optional uint64 sequence = 5;
+
+	optional uint64 ttl = 6;
```

Comment:
Is this taken from go-ipfs? What does TTL mean here? Does the IPNS implementation track when it receives records so that it knows when to purge them?

Reply:
This proto definition has to be the same as go, in order to guarantee the interoperability among them. 

The `ttl` parameter will also be used in the IPNS for caching purposes, as you can see [here](https://github.com/ipfs/go-ipfs/blob/master/core/commands/publish.go#L73). But since it is still experimental in go, I decided to leave it with low priority for now.